### PR TITLE
Fixes for timeline semaphore

### LINKED ISSUE #588: Timeline semaphore sample not available on Windows
Due to a kernel bug the timeline semaphore sample was disabled on windows:

> Not enabled on Windows at this time due to bugs.
Out-of-order submission in presentation causes kernel level issues,
and need to be figured out before this sample can be enabled on Windows.

We should reevaluate this and if it works on Windows now, the sample should be enabled for that platform again.

### PR DESCRIPTION
**Pull request by bryce-young-mobica**
## Description
Reworked the timeline semaphore sample to prevent it crashing on Windows. 

The only obvious trigger for the crash that I could observe was the main thread calling "vkDeviceWaitIdle" whilst the compute thread was in "wait_timeline_gpu". To avoid this, I removed the "wait/signal_timeline_gpu" calls (opting to attach the "VkTimelineSemaphoreSubmitInfo" to the queue submissions instead), and restructured the compute/graphics work stages to prevent the compute thread running ahead and submitting (potentially blocking) work.

- Replaced the "wait/signal_timeline_gpu" calls with "VkTimelineSemaphoreSubmitInfo" added to the queue submissions
- Added an enum for identifying the timeline stages, to make it easier to read and debug the signals/waits
- Moved the rendering work to a seperate thread, leaving the main thread responsible for acquiring and presenting the swapchain image, and signalling the timeline semaphore.

Fixes #588

Tested on Windows
Tested with shared queue for compute and graphics
## General Checklist:

Please ensure the following points are checked:

- [X] My code follows the [coding style](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/CONTRIBUTING.adoc#Code-Style)
- [X] I have reviewed file [licenses](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/CONTRIBUTING.adoc#Copyright-Notice-and-License-Template)
- [X] I have commented any added functions (in line with Doxygen)
- [X] I have commented any code that could be hard to understand
- [X] My changes do not add any new compiler warnings
- [] My changes do not add any new validation layer errors or warnings
- [X] I have used existing framework/helper functions where possible
- [X] My changes do not add any regressions
- [X] I have tested every sample to ensure everything runs correctly
- [X] This PR describes the scope and expected impact of the changes I am making

 Note: The Samples CI runs a number of checks including:
 - [X] I have updated the header Copyright to reflect the current year (CI build will fail if Copyright is out of date)
 - [X] My changes build on Windows, Linux, macOS and Android. Otherwise I have [documented any exceptions](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/CONTRIBUTING.adoc#General-Requirements)

 If this PR contains framework changes:
 - [X] I did a full batch run using the `batch` command line argument to make sure all samples still work properly

## Sample Checklist

If your PR contains a new or modified sample, these further checks must be carried out *in addition* to the General Checklist:
- [X] I have tested the sample on at least one compliant Vulkan implementation
- [X] If the sample is vendor-specific, I have [tagged it appropriately](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/CONTRIBUTING.adoc#General-Requirements)
- [X] I have stated on what implementation the sample has been tested so that others can test on different implementations and platforms
- [X] Any dependent assets have been merged and published in downstream modules
- [X] For new samples, I have added a paragraph with a summary to the appropriate chapter in the readme of the folder that the sample belongs to [e.g. api samples readme](https://github.com/KhronosGroup/Vulkan-Samples/blob/main/samples/api/README.adoc)
- [X] For new samples, I have added a tutorial README.md file to guide users through what they need to know to implement code using this feature. For example, see [conditional_rendering](https://github.com/KhronosGroup/Vulkan-Samples/tree/main/samples/extensions/conditional_rendering)
- [X] For new samples, I have added a link to the [Antora navigation](https://github.com/KhronosGroup/Vulkan-Samples/blob/main/antora/modules/ROOT/nav.adoc) so that the sample will be listed at the Vulkan documentation site

### CODE REVIEW

File: samples/extensions/timeline_semaphore/CMakeLists.txt
Comment (SaschaWillems) [inline samples/extensions/timeline_semaphore/CMakeLists.txt:1]: Looks like all line endings were changed with this PR, making it look like this whole file changed. That makes it hard to review. Can you revert the line ending changes?

File: samples/extensions/timeline_semaphore/README.adoc
Comment (SaschaWillems) [inline samples/extensions/timeline_semaphore/README.adoc:1]: Looks like all line endings were changed with this PR, making it look like this whole file changed. That makes it hard to review. Can you revert the line ending changes?

**Review by SaschaWillems — CHANGES_REQUESTED** [review]

File: samples/extensions/timeline_semaphore/CMakeLists.txt
Comment (bryce-young-mobica) [reply samples/extensions/timeline_semaphore/CMakeLists.txt:1]: Sorry, meant to create the PR for a forked repo (haven't used github much). Thanks for the review though, I'll resubmit this PR once from our fork when it's ready

**Comment by SaschaWillems** [comment]
Any update on this? Would be great to have that sample properly work on Windows.

**Comment by bryce-young-mobica** [comment]
Sorry for the delay. 
I made the changes you suggested, and opened a new PR here: https://github.com/KhronosGroup/Vulkan-Samples/pull/1125 (didn't see the reopen option until after I'd created it).

**Pull request closed without merging**